server/lib/fusor/password_filter.rb

```diff
@@ -0,0 +1,85 @@
+##
+# PasswordFilter
+# ================
+# Extracts passwords from deployment objects.
+# Provides functionality for filtering sensitive data (passwords) from text.
+
+class PasswordFilter
+  def self.password_cache
+    return @password_cache
+  end
+
+  def self.extract_deployment_passwords(deployment)
+    # act as passthrough (no filtering) if we're in dev/test AND show_passwords is enabled
+    if !Rails.env.production? and SETTINGS[:fusor][:system][:logging][:show_passwords]
+      return nil
+    end
+
+    main_deployment_passwords = [
+      :rhev_engine_admin_password,
+      :rhev_root_password,
+      :rhev_gluster_root_password,
+      :cfme_root_password,
+      :cfme_admin_password,
+      :cfme_db_password,
+      :openshift_user_password,
+      :openshift_root_password,
+    ]
+    osp_deployment_passwords = [
+      :undercloud_admin_password,
+      :undercloud_ssh_password,
+      :overcloud_password,
+    ]
+
+    # get passwords from main deployment
+    extracted_passwords = Set.new
+    extracted_passwords += cautious_get_attrs(main_deployment_passwords, deployment)
```

Comment:
Good idea on compressing to one line! Slipped my mind (still learning Ruby).

Do you think there is a performance difference for Set vs Array here? Set seemed like the most natural fit due to the nature of the data contained.